test/extensions/filters/http/cache/cache_headers_utils_test.cc

```diff
@@ -518,161 +529,261 @@ TEST(GetAllMatchingHeaderNames, MultipleMatches) {
+std::string GetParseCommaDelimitedHeaderTestName(
+    const testing::TestParamInfo<ParseCommaDelimitedHeaderTestCase>& info) {
+  return std::string(info.param.name);
+}
+
+std::vector<ParseCommaDelimitedHeaderTestCase> ParseCommaDelimitedHeaderTestParams() {
+  return {
+      {
+          "Null",
+          {},
+          {},
+      },
+      {
+          "Empty",
+          {},
+          {},
+      },
+      {
+          "SingleValue",
+          {"accept"},
+          {"accept"},
+      },
+      {
+          "MultiValue",
+          {"accept,accept-language"},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiValueLeadingSpace",
+          {" accept,accept-language"},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiValueSpaceAfterValue",
+          {"accept ,accept-language"},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiValueTrailingSpace",
+          {"accept,accept-language "},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiValueLotsOfSpaces",
+          {"  accept  ,  accept-language  "},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiEntry",
+          {"accept", "accept-language"},
+          {"accept", "accept-language"},
+      },
+      {
+          "MultiEntryMultiValue",
+          {"accept,accept-language", "foo,bar"},
+          {"accept", "accept-language", "foo", "bar"},
+      },
+      {
+          "MultiEntryMultiValueWithSpaces",
+          {"accept,  accept-language  ", "foo  ,bar"},
+          {"accept", "accept-language", "foo", "bar"},
+      },
+  };
+}
+
+class ParseCommaDelimitedHeaderTest
+    : public testing::TestWithParam<ParseCommaDelimitedHeaderTestCase> {};
+
+INSTANTIATE_TEST_SUITE_P(ParseCommaDelimitedHeaderTest, ParseCommaDelimitedHeaderTest,
+                         testing::ValuesIn(ParseCommaDelimitedHeaderTestParams()),
+                         GetParseCommaDelimitedHeaderTestName);
+
+TEST_P(ParseCommaDelimitedHeaderTest, ParseCommaDelimitedHeader) {
+  ParseCommaDelimitedHeaderTestCase test_case = GetParam();
+  const Http::LowerCaseString header_name = Http::CustomHeaders::get().Vary;
+  Http::TestResponseHeaderMapImpl headers;
+  for (absl::string_view entry : test_case.header_entries) {
+    headers.addCopy(header_name, entry);
+  }
+  std::vector<absl::string_view> result =
+      CacheHeadersUtils::parseCommaDelimitedHeader(headers.get(header_name));
+  std::vector<absl::string_view> expected(test_case.expected_values.begin(),
+                                          test_case.expected_values.end());
+  EXPECT_EQ(result, expected);
 }
 
-TEST(ParseCommaDelimitedList, Empty) {
-  Http::TestResponseHeaderMapImpl headers{{"vary", ""}};
-  std::vector<std::string> result =
-      CacheHeadersUtils::parseCommaDelimitedList(headers.get(Http::CustomHeaders::get().Vary));
+TEST(CreateVaryIdentifier, IsStableForAllowListOrder) {
```

Comment:
Could you add a test for when a vary header isn't in the white-list (unless I missed it)?

Reply:
Apparently directly linking to a file that's hidden by default doesn't work very well...

The test is this one, L774:
```
TEST(CreateVaryIdentifier, DisallowedHeader)
```